### ISSUE TITLE
Crashing in The Witcher 1 (exception_access_violation)

### ISSUE THREAD

**Issue by vessbon** [doitsujin/dxvk#3682]
The game crashes whenever I load a save, try to create a new game, loading into new areas and sometimes just randomly crashes.
I have had a long history of crashes with this game, recently reinstalled windows not only because of this but I hoped it would be a nice side effect. First playthrough went smoothly, I wanted to play it again with mods and dxvk for increased performance but this makes it unbearable.

In case you are wondering, I used this reddit post [here](https://www.reddit.com/r/witcher/comments/kg3u76/the_witcher_enhanced_edition_revised_modding_guide/) install mods properly. I skipped some mods so I wouldn't have to crash because of anything memory related but of course that didn't want to work.

### Software information
The Witcher: Enhanced Edition (V1.5A).
Maxed out settings.

### System information
- GPU: RX 7900XT
- Driver: 23.9.3
- Wine version: None (Windows 11 22H2 x64)
- DXVK version: 2.3

### Log files
Game Log:
[errlog.txt](https://github.com/doitsujin/dxvk/files/12784323/errlog.txt)

DXVK D3D9 Log:
[witcher_d3d9.log](https://github.com/doitsujin/dxvk/files/12784328/witcher_d3d9.log)

**Comment by Blisto91** [comment]
Can you list which of the mods you are using.

**Comment by vessbon** [comment]
I am using FCR, Complete Sword Overhaul, Dagger Overhaul, AI Upscaled Textures (AiO AI Textures with RotWW and The Witcher Overhaul Project by lexo1000), The Witcher Overhaul Project (Only sound and blood files), Celin Wraith Fix, Fair Kalkstein Mod, Plentiful Herbs, Han Gives Han NOT Berbercane, Reverend Beard Texture Fix, Faster Movement For Geralt.

**Comment by vessbon** [comment]
I have also patched the game to be Large Address Aware using FPS Counter which I forgot to mention.

**Comment by K0bin** [comment]
Please make an apitrace.

**Comment by vessbon** [comment]
> Please make an apitrace.

Trying to launch the game with the apitrace d3d9.dll crashes the game when I start it. I used the 32 bit files.

**Comment by vessbon** [comment]
I tried doing everything I can to get the apitrace to work, are there any known issues with it or do I just suck?

**Comment by Blisto91** [comment]
Sometimes it just doesn't work with some games.

I will take a look later when i have the time.

**Comment by vessbon** [comment]
Yeah I saw that it might not work in some games and that's what I thought was happening to begin with but to make sure I reinstalled the game and tried without mods and a lot of different stuff, nothing worked so I guess it is the case after all.
I think you and K0bin have handled a very similar issue before about a year back: https://github.com/doitsujin/dxvk/issues/1763#issuecomment-1152942503

Maybe Renderdoc would be useful?

**Comment by K0bin** [comment]
> Maybe Renderdoc would be useful?

No, it wouldn't. Renderdoc captures aren't portable so they can only be replayed on a similar GPU and driver as the one you have. On top of that they only show one frame which is probably not the frame the game crashes in anyway.

**Comment by vessbon** [comment]
So basically dxvk just won't work with The Witcher 1 modded?

**Comment by K0bin** [comment]
Maybe I'll look into it at some point but I simply don't have time to mess with Witcher 1 mods at the moment.

**Comment by mirh** [comment]
Putting aside that I wouldn't really put my horses on a guide that seems like recommending v-sync as a shorthand for fps limiting.. does the thing work just fine with just native d3d9?
Because once you start to mod everything and the kitchen sink it may just simply be that the game has not enough address space. 
(and maybe dxvk-remix could still save you there, but this is besides the point)

**Comment by Blisto91** [comment]
@vessbon Using just the mods you mentioned i haven't had it crash upon a new game yet. But it does happen when i use the fps counter tool at least on Linux.
Does it stop crashing if you remove that?

**Comment by Blisto91** [comment]
Apitraces of the above crash when using fps counter. One with wined3d where it doesn't crash and one with dxvk where it does.
https://drive.proton.me/urls/1G9ATM1Y50#TOZZq67EFwz3

**Comment by K0bin** [comment]
UPDATE: I can reproduce the issue with a GCC build of DXVK. The problem is that running the game with GDB makes it work fine and Visual Studio doesn't have symbols for the GCC build.

So I have no idea how to further investigate this.

**Comment by rhazor** [comment]
I don't know if this is relevant, but the game, Witcher 1, crashes with dxvk when loading saves or changing zones if running any software that injects any kind of hook, e.g. the game crashes almost 100% of the time with OBS Studio, also I think it crashes with RTSS.

If you close OBS, RTSS etc. but keep dxvk - the crashes stop.
If you do vice versa, remove dxvk, but keep OBS, RTSS, etc, - the crashes also stop.
Go figure.

Also you don't need to record or stream with OBS, just leave it open with game capture set to the witcher.exe

I do run same/similar mods, all tagged [ MANDATORY ] from this reddit guide [here](https://www.reddit.com/r/witcher/comments/kg3u76/the_witcher_enhanced_edition_revised_modding_guide/).

Also I had same exact issues with Oblivion dxvk+OBS, crashes were less common tho, but on The Witcher 1 it's unplayable.
